pandas_transform/pandas_transform.md

```diff
@@ -0,0 +1,995 @@
+
+@sage
+<div class="python_data_init">
+<lia-keep>
+<script type="text/x-sage">
+import pandas as pd
+covid_testing = pd.read_csv('https://raw.githubusercontent.com/arcus/education_modules/embedded_code/a_sample_module_template/covid_testing.csv')
+</script>
+</lia-keep>
+</div>
+<br>
+
+You come across the following code in which several conditions are defined but not given descriptive names:
+
+<div class="python_data_init">
+<lia-keep>
+<script type="text/x-sage">
+
+condition_1 = covid_testing.loc[:,"first_name"].isna()
+
+condition_2 = covid_testing.loc[:,"last_name"].isna()
+
+condition_3 = covid_testing.loc[:,"age"]<=18
+
+my_condition = (condition_1 | condition_2) & condition_3
+</script>
+</lia-keep>
+</div>
+<br>
+
+<div class="python_data">
+<lia-keep>
+<script type="text/x-sage">
+
+</script>
+</lia-keep>
+</div>
+<br>
+
+What does `my_condition` test for? You can use the interactive cell above to test the conditions however you want.
+
+[( )] Patients under the age of 18 who are missing data in both the `first_name` and `last_name` columns.
+[( )] Patients under the age of 18 who are missing data in either the `first_name` or `last_name` columns.
+[( )] Patients ages 18 or younger who are missing data in both the `first_name` and `last_name` columns.
+[(X)] Patients ages 18 or younger who are missing data in either the `first_name` or `last_name` columns.
+***
+<div class = "answer">
+* `condition_1` will be TRUE if the patient is missing data in the `first_name` column.
+
+* `condition_2` will be TRUE if the patient is missing data in the `last_name` column.
+
+* `condition_1 | condition_2` will be TRUE if the patient is missing data in the `first_name` column, the `last_name` column, or in both columns.
+
+* `condition_3` will be TRUE if the patient is age 18 or less.
+
+THEREFORE:
+
+* `my_condition` will be TRUE when **both** `condition_1 | condition_2` and `condition_3` are TRUE, when **at least one** of the columns `first_name` and `last_name` are blank, and the patient age is 18 or less.
+
+</div>
+***
+
+## Transforming data in a DataFrame
+
+Sometimes the data in a DataFrame isn't in the best form for us to use. Maybe it is a string that is too long or unclear, or maybe the units aren't the unit we will ultimately want to use to analyze the data.
+
+There are two things we can do to make our data more useful.
+
+1. Add a new column with data based on existing columns.
+
+2. Edit the data in an existing column to improve its usefulness.
+
+<div class ="warning">
+Reproducibility must always been a priority when doing research, so whenever you are making changes to data, you must ensure that you keep a clear record of the calculations you did and even more importantly that you don't change the original copy of your data.
+
+**Recording calculations**
+
+Python, and particularly Jupyter notebooks, in which you can write linked cells of code, are a great way to preserve and annotate every step of your data manipulation.
```

Comment:
Wondering if you want to rework to not highlight jupyter here, since they're not actually using it. I'm okay if you just leave it, though.